### PR TITLE
gps add device_id

### PR DESCRIPTION
**Pull request by dagar**
Adding device id to GPS so that metadata like position offset or estimator tuning can be properly configured (and persist) for specific modules. It's also helpful to identify what's connected at a glance.

~~Requires https://github.com/PX4/PX4-GPSDrivers/pull/79~~.

Screen Shot 2021-04-01 at 11 14 36 PM

### CODE REVIEW

**Review by LorenzMeier — DISMISSED** [review]

File: src/drivers/gps/gps.cpp [hunk 1]
```diff
@@ -987,6 +1004,42 @@ void
...
+					break;
+
+				case GPSDriverUBX::Board::u_blox9_F9P:
+					set_device_type(DRV_GPS_DEVTYPE_UBX_F9P);
+					break;
+
+				default:
+					set_device_type(DRV_GPS_DEVTYPE_UBX);
+					break;
+				};
```
Comment (bkueng) [inline src/drivers/gps/gps.cpp:1037]:
```suggestion
				}
```

File: src/drivers/gps/gps.cpp [hunk 2]
```diff
@@ -987,6 +1004,42 @@ void
 GPS::publish()
 {
 	if (_instance == Instance::Main || _is_gps_main_advertised.load()) {
+
+		if (_mode == GPS_DRIVER_MODE_UBX) {
+			// populate specific ublox model
```
Comment (bkueng) [inline src/drivers/gps/gps.cpp:1009]: Can you move this to https://github.com/PX4/PX4-Autopilot/blob/15c230d18af7bf2de275cfb4adf907a3b4edf942/src/drivers/gps/gps.cpp#L796?

Comment (dagar) [reply src/drivers/gps/gps.cpp:1009]: Done.

File: src/drivers/gps/gps.cpp (hunk 1 above)
Comment (dagar) [reply src/drivers/gps/gps.cpp]: Fixed.

**Review by bkueng — APPROVED** [review]

**Pull request merged**